lib/jobs/ucs-discovery.js

```diff
@@ -0,0 +1,169 @@
+            })
+            .then(function() {
+                self._done();
+            })
+            .catch(function(err) {
+                self._done(err);
+            });
+    };
+
+    /**
+     * @function logIn
+     * @description Log into the UCS microservice
+     */
+    UcsDiscoveryJob.prototype.logIn = function () {
+        var username = this.settings.username;
+        var password = this.settings.password;
+        var host = this.settings.ucs;
+        var url= "/login?host=" + host+ "&user="+ username + "&password=" + password;
+        var d ={};
+        d.username=this.settings.username;
+        d.password=this.settings.password;
+        return this.ucs.clientRequest(url)
+            .then(function(response) {
+                return response.body;
+            });
+    };
+
+
+    /**
+     * @function getRoot
+     * @description return all the elments in the Cisco UCS that RackHD cares about
+     */
+    UcsDiscoveryJob.prototype.getRoot = function () {
+        var username = this.settings.username;
+        var password = this.settings.password;
+        var ucs = this.settings.ucs;
+        var baseurl= this.settings.protocol+"://"+this.settings.host+ ":"+this.settings.port;
+        var url= baseurl = "/sys?host=" + ucs+ "&user="+ username + "&password=" + password;/* jshint ignore:line */
+
+        return this.ucs.clientRequest(url)
+            .then(function(response) {
+                return response.body;
+            });
+    };
+
+    /**
+     * @function createRackmounts
+     * @description discovers all the rackmount servers in the Cisco UCS
+     */
+    UcsDiscoveryJob.prototype.createRackmounts = function (root) {
+        var username = this.settings.username;
+        var password = this.settings.password;
+        var ucs = this.settings.ucs;
+
+        if (!_.has(root, 'Servers')) {
+            logger.warning('No rackmount servers found Found');
+            return Promise.resolve();
+        }
+        var baseurl= this.settings.protocol+"://"+this.settings.host+ ":"+this.settings.port;
+        var url= baseurl = "/rackmount?host=" + ucs+ "&user="+ username + "&password=" + password;/* jshint ignore:line */
+        return this.ucs.clientRequest(url)
+            .then(function(res) {
+                assert.object(res);
+                return res.body;
+            })
+            .map(function(data) {
+                assert.array(data.macs);
+                assert.string(data.name);
+                assert.string(data.path);
+                var node = {
+                    type: 'compute',
+                    name: data.name,
+                    identifiers: data.macs,
+                    obm:[],
+                    relations : []
+                };
+                var obm = {
+                    config: "",
+                    service: 'ucs-obm-service'
+                };
+                node.obm.push(obm);
```

Comment:
@tannoa2 No need to do a push here.  You can declare `obm` before `node` and set use `obm: [ obm ]` to set the obm property.